### PR TITLE
#196754 block unblock user

### PR DESCRIPTION
**Pull request by snehalsabade**
Sir please review PR of block unblock user 
PR: https://github.com/tekdi/sunbird-admin-ui/pull/14

When -User click on delete button
If - User is an active 
Then - Show confirm dialog want to sure block user with confirmtion 

When -User click on delete button
If - User is an inactive 
Then - Show confirm dialog want to sure an unblock user with yes or no confirmation

Implementtion Video: https://www.loom.com/share/cd6fc7ec30be453a922a67e72ecf9bb2

### CODE REVIEW

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 1]
```diff
@@ -63,12 +63,35 @@ <h5 class="m-0">{{ 'COMMON_USER' | i18next}}</h5><br>
...
-                                <button pButton pRipple icon="pi pi-trash" class="p-button-rounded p-button-warning"></button>
+                                <button pButton pRipple icon="pi pi-trash" class="p-button-rounded p-button-warning" (click)="blockUser(user)" *ngIf="user.status"></button>
+                                <button pButton pRipple icon="pi pi-unlock" class="p-button-rounded p-button-warning" (click)="unblockUser(user)" *ngIf="!user.status"></button>
                             </div>
                         </td>
                     </tr>
                 </ng-template>
             </p-table>
+
+            <p-dialog [(visible)]="deleteUserDialog" header="Confirm" [modal]="true" >
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: @snehalsabade can we use only one p-dialog for block and unblock user

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts [hunk 2]
```diff
@@ -147,5 +148,46 @@ export class SbUserComponent implements OnInit {
             height: 'auto'
         });
     }
-}
 
+  blockUser(user: User) {
+    this.deleteUserDialog = true;
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.ts]: Either use "delete" or "block" words. 
deleteUserDialog => blockUserDialog
unblockUserDialog

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts [hunk 3]
```diff
@@ -147,5 +148,46 @@ export class SbUserComponent implements OnInit {
...
+    this.user = user;
+  }
+  confirmBlock() {
+    const payload = {
+      "request": {
+        "userId": this.user.userId
+      }
+    }   
+    this.userService.blockUser(payload).subscribe(response => {
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.ts]: merge block and unblock into one function

File: src/app/sb-admin/service/user.service.ts [hunk 4]
```diff
@@ -40,4 +40,22 @@ export class UserService {
     })
     return this.http.post(environment.userCreateUrl, payload, { headers: header });
   }
+
+  blockUser(payload: any): Observable<any> {
```
Comment (gouravmore) [inline src/app/sb-admin/service/user.service.ts]: use only one service to block and unblock

File: src/locales/en/en.json [hunk 5]
```diff
@@ -60,5 +60,11 @@
   "USER_SAVE":"Save",
   "USER_ROLES":" Roles",
   "USER_ROLE_ADDED":"User Role Updated Successfully",
-  "COMMON_USER":"Users"
+  "COMMON_USER":"Users",
```
Comment (gouravmore) [inline src/locales/en/en.json]: USER or USERS?

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 1 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: resolved

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts (hunk 2 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.ts]: resolved

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts (hunk 3 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.ts]: resolved

File: src/app/sb-admin/service/user.service.ts (hunk 4 above)
Comment (snehalsabade) [reply src/app/sb-admin/service/user.service.ts]: resolved

File: src/locales/en/en.json (hunk 5 above)
Comment (snehalsabade) [reply src/locales/en/en.json]: resolved

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 1 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: yes sir, now showing only one p-dialog for block and unblock user

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 6]
```diff
@@ -63,12 +63,26 @@ <h5 class="m-0">{{ 'COMMON_USER' | i18next}}</h5><br>
                         <td>
                             <div class="flex">
                                 <button [disabled]="!user.status" pButton pRipple icon="pi pi-pencil" class="p-button-rounded p-button-info mr-2" (click)="editRole(user)"></button>
-                                <button pButton pRipple icon="pi pi-trash" class="p-button-rounded p-button-warning"></button>
+                                <button pButton pRipple icon="pi pi-trash" class="p-button-rounded p-button-warning" (click)="blockUnblockUser(user)" *ngIf="user.status"></button>
+                                <button pButton pRipple icon="pi pi-unlock" class="p-button-rounded p-button-warning" (click)="blockUnblockUser(user)" *ngIf="!user.status"></button>
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: Use one button to show different icons.

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 7]
```diff
@@ -63,12 +63,26 @@ <h5 class="m-0">{{ 'COMMON_USER' | i18next}}</h5><br>
...
                             </div>
                         </td>
                     </tr>
                 </ng-template>
             </p-table>
+
+            <p-dialog [(visible)]="blockUnblockUserDialog" header="Confirm" [modal]="true" >
+                <div class="flex align-items-center justify-content-center">
+                    <i class="pi pi-exclamation-triangle mr-3" style="font-size: 2rem"></i>
+                    <span *ngIf="user?.status === 1 ">{{'USER_BLOCK_CONFIRMATION' | i18next}} <b>{{user?.firstName}}</b>?</span>
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: Use only one span to show status

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 8]
```diff
@@ -63,12 +63,26 @@ <h5 class="m-0">{{ 'COMMON_USER' | i18next}}</h5><br>
...
+                    <span *ngIf="user?.status === 0" >{{'USER_UNBLOCK_CONFIRMATION' | i18next}} <b>{{user?.firstName}}</b>?</span>
+                </div>
+                <ng-template pTemplate="footer">
+                    <button pButton pRipple icon="pi pi-times" class="p-button-text" label="{{'COMMON_NO' | i18next}}" (click)="blockUnblockUserDialog = false"></button>
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html:80]: Use only one button tag

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts [hunk 9]
```diff
@@ -147,5 +146,30 @@ export class SbUserComponent implements OnInit {
...
+    this.user = user;
+  }
+  confirmBlock() {
+    const payload = {
+      "request": {
+        "userId": this.user.userId
+      }
+    }
+    this.userService.blockUnblockUser(payload, this.user?.status).subscribe(response => {
+      this.messages = [];
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.ts:164]: what is the use of this statement?

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts [hunk 10]
```diff
@@ -147,5 +146,30 @@ export class SbUserComponent implements OnInit {
...
+      if (this.user.status) {
+        this.user.status = 0;
+        this.messageService.add({ severity: 'success', detail: this.i18nextPipe.transform('USER_BLOCK_SUUCCESSFULLY') });
+      } else {
+        this.user.status = 1;
+        this.messageService.add({ severity: 'success', detail: this.i18nextPipe.transform('USER_UNBLOCK_SUUCCESSFULLY') });
+      }
+      this.blockUnblockUserDialog = false;
+    }, (error: any) => {
+      this.messages = [];
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.ts:174]: what is the use of this statement?

File: src/app/sb-admin/service/user.service.ts [hunk 11]
```diff
@@ -40,4 +40,17 @@ export class UserService {
...
   }
+
+  blockUnblockUser(payload: any, userStatus: any): Observable<any> {
+    let header = new HttpHeaders({
+      "Content-Type": 'application/json',
+      "Authorization": environment.authKey,
+      "x-authenticated-user-token": environment.userToken,
+    })
+    if (userStatus) {
+      return this.http.post(environment.blockUserUrl, payload, { headers: header });
```
Comment (gouravmore) [inline src/app/sb-admin/service/user.service.ts:51]: Is block and unblock have different urls?

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/service/user.service.ts (hunk 11 above)
Comment (snehalsabade) [reply src/app/sb-admin/service/user.service.ts:51]: yes sir

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 7 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: resolved

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 6 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: resolved

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts (hunk 9 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.ts:164]: empty the previous message and show the current one

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.ts (hunk 10 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.ts:174]: empty the previous message and show the current one

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 8 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html:80]: we are showing two button on a single dialog at a time.

File: src/app/sb-admin/service/user.service.ts (hunk 11 above)
Comment (gouravmore) [reply src/app/sb-admin/service/user.service.ts:51]: These URLs should come from url configs json. I think this is resolved in another PR.

**Review by gouravmore — APPROVED** [review]

**Pull request merged**